core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala

```diff
@@ -625,7 +626,7 @@ class DynamicBrokerReconfigurationTest extends QuorumTestHarness with SaslSetup
       "Config not updated in LogManager")
 
     val log = servers.head.logManager.getLog(new TopicPartition(topic, 0)).getOrElse(throw new IllegalStateException("Log not found"))
-    TestUtils.waitUntilTrue(() => log.config.segmentSize == 4000, "Existing topic config using defaults not updated")
+    TestUtils.waitUntilTrue(() => log.config.segmentSize == 1048576, "Existing topic config using defaults not updated")
```

Comment:
You will need to change validation in line 641 as well to retain the original intention of this test `log.logSegments.asScala.exists(_.size > 3000)`

Reply:
I will have to improve the produce/consume part of this test to fill a 1MB segment. I will have a go at this.